### PR TITLE
Disable Multiple login and signup button presses.

### LINKED ISSUE #178: Prevent multiple button presses for login & sign up
Users should be prevented from repeatedly pressing the button to login or sign to the app so that data is not being sent multiple times causing authentication issues.

Depends on #13 & #14

### PR DESCRIPTION
**Pull request by HenderikusP**
**Describe the issue**
Users should be prevented from repeatedly pressing the button to login or sign to the app so that data is not being sent multiple times causing authentication issues.

**Describe the solution**
created a state where if the auth was approved, the button was replaced by a button without a click functionality.

### CODE REVIEW

**Review by sannan06 — COMMENTED** [review]
Linter issues need addressing

**Review by Denijar — APPROVED** [review]
Looks good to me :)

File: src/pages/loginView.js [hunk 1]
```diff
@@ -44,6 +44,7 @@ export default function LoginView() {
     }
 
     if (validLogin) {
+      setSuccessfulLogin(true);
```
Comment (dikunong) [inline src/pages/loginView.js:47]: Once this runs, it never flips back, meaning that if login fails for any reason (e.g. incorrect password) the user is locked out unless they refresh the page. Running `setSuccessfulLogin(false)` on line 55 should fix that I think :)

File: src/pages/signUpView.js [hunk 2]
```diff
@@ -63,6 +64,7 @@ export default function SignUpView() {
     }
 
     if (validSignup) {
+      setSuccessfulSignUp(true);
```
Comment (dikunong) [inline src/pages/signUpView.js:67]: It's much less likely to happen on sign up, but it's also possible to be locked out here. Similarly to the login page, running `setSuccessfulSignUp(false)` on line 82 should fix that.

**Review by dikunong — CHANGES_REQUESTED** [review]
Good work! There's a bug where if you fail to login/signup on the first go (e.g. if you typed your password wrong) it'll lock the user out permanently - my comments detail the changes needed that should fix it.

**Review by dikunong — APPROVED** [review]
LGTM, cheers for being onto it with fixing that bug :)

**Pull request merged**